lib/improver/utilities/cube_manipulation_new.py

```diff
@@ -312,363 +314,372 @@ def concatenate_cubes(
     return result
 
 
-def merge_cubes(cubes, model_id_attr=None, blend_coord=None):
+class MergeCubes():
     """
-    Function to merge cubes, accounting for differences in the
-    attributes, and coords.
+    Class adding functionality to iris.merge_cubes()
 
-    Args:
-        cubes (Iris cubelist or Iris cube):
-            Cubes to be merged.
+    Accounts for differences in attributes and coordinates to avoid merge
+    failures and anonymous dimensions.
+    """
+    def __init__(self):
+        """Initialise constants"""
+        self.silent_attributes = ["history", "title", "mosg__grid_version"]
+        self.coord_mismatch_error_keys = ["threshold"]
 
-    Kwargs:
-        model_id_attr (str):
-            Name of cube attribute used to identify the model for grid
-            blending or None.
-        blend_coord (str):
-            Name of coordinate over which merged cube is to be blended, or
-            None. If "time", triggers bounds range checks to avoid passing
-            eg mismatched accumulation periods into blending.
+    def _equalise_cubes(self, cubelist):
+        """
+        Function to equalise cubes where they do not match.
 
-    Returns:
-        result (Iris cube):
-            Merged cube.
+        Args:
+            cubelist (iris.cube.CubeList):
+                List of cubes to check and equalise.
 
-    """
-    if isinstance(cubes, iris.cube.Cube):
-        cubes = iris.cube.CubeList([cubes])
+        Returns:
+            cubelist (iris.cube.CubeList):
+                List of cubes with revised cubes. The number of cubes in this
+                list may be greater than the original number of cubes if they
+                have been sliced over mismatching dimension coordinates.
+        """
+        equalise_cube_attributes(cubelist, silent=self.silent_attributes)
+        strip_var_names(cubelist)
 
-    cubelist = _equalise_cubes(
-        cubes, model_id_attr=model_id_attr, merging=True)
+        cubelist = self._equalise_cube_coords(cubelist)
+        cubelist = self._equalise_cell_methods(cubelist)
+        for cube in cubelist:
+            check_cube_not_float64(cube, fix=True)  # TODO why here?
```

Comment:
Yes, I wasn't envisaging changing this at the moment, but I think it's worth leaving the TODO as we we will want to fix this - it's a bad idea to have a type cast buried where it might cause problems and be difficult to find.  This probably links in with the metadata and datatype conformance work.